python/ray/tests/test_runtime_env_conda_and_pip_2.py

```diff
@@ -62,5 +62,114 @@ def f():
         lambda: bad_packages["task"] in capsys.readouterr().out, timeout=30)
 
 
+@pytest.mark.skipif(
+    os.environ.get("CI") and sys.platform != "linux",
+    reason="Requires PR wheels built in CI, so only run on linux CI machines.")
+@pytest.mark.parametrize("field", ["conda", "pip"])
+@pytest.mark.parametrize("spec_format", ["file", "python_object"])
+def test_actor_level_gc(start_cluster, field, spec_format, tmp_path):
+    """Tests that actor-level working_dir is GC'd when the actor exits."""
```

Comment:
Can you add one more python test to test worker registration timeout?

Reply:
I have covered this case in worker pool test. By the way, it is not easy to simulate this in python.

Reply:
You can simulate this with runtime env I think.

Reply:
Let me see if I can find an example. I am sure there’s a test that simulates thjd

Reply:
https://github.com/ray-project/ray/blob/ed344341316f29b0c96f9579923a8c106374e04c/python/ray/tests/test_failure_4.py#L440

Reply:
I understand this is covered by cpp test. One concern I have is our worker pool API is not so great that I am not confident it actually conveys the real world workflow (e.g., It is possible some of API is not called on the right order in some scenarios).

Reply:
So, Do you think I should remove the test from worker pool if we already cover it in python? Actually, I didn't add tests in worker pool before? But @edoakes comment me to add the cpp test https://github.com/ray-project/ray/pull/20789#pullrequestreview-820922744.

Reply:
Why not just adding both?

Reply:
Oh, python worker can't simulate worker process with multiple workers.

Reply:
Oh I meant worker registration failure scenario, not multiple workers

Reply:
OK, let me see.

Reply:
Add a test here https://github.com/ray-project/ray/pull/20789/commits/884188c1b27d58e412b2f172fd5d7cb7c3e704eb#diff-9a58a116bef057db388a6eef0a46fc55521de1bc9568781a5606792fea7de782R60